src/Analyzer.Utilities/DiagnosticHelpers.cs

```diff
@@ -98,5 +99,117 @@ public static string GetMemberName(ISymbol symbol)
             // For other language constructs
             return symbol.Name;
         }
+               
+        public static bool IsInvisibleOutsideAssemblyAtSymbolLevel(ISymbol symbol)
+        {
+            return SymbolIsPrivateOrInternal(symbol)
+                || SymbolIsProtectedInSealed(symbol);
```

Comment:
This is probably incorrect. For example, consider a public type defined inside an internal type. See `GetResultantVisibility` method in ISymbolExtensions.cs